staging/src/k8s.io/client-go/plugin/pkg/client/auth/exec/exec.go

```diff
@@ -212,6 +217,27 @@ func newAuthenticator(c *cache, config *api.ExecConfig, cluster *clientauthentic
 	return c.put(key, a), nil
 }
 
+func isInteractive(isTerminalFunc func(int) bool, config *api.ExecConfig) (bool, error) {
```

Comment:
Open question: how do we ensure that a higher level entity is not using standard input?

I think this is hard to do. Some discussion here:
- https://github.com/kubernetes/kubernetes/issues/98451#issuecomment-783501007
- https://kubernetes.slack.com/archives/C0EN96KUY/p1613768319115600?thread_ts=1613767745.112500&cid=C0EN96KUY

Reply:
I am wondering if could probe `os.Stdin` somehow to tell if `kubectl` is already using it for something else (in `UpdateTransportConfig` or `getUserIdentificationPartialConfig` or ...).

TODO: look into this more next week.

Reply:
I also have been looking into similar credential plugins to see if there is any prior art here. TLDR: I can't find anyone else that is hooking up a TTY to stdin on the credential plugin.
- Docker credential plugins (takes command input on stdin): https://docs.docker.com/engine/reference/commandline/login/#credential-helpers
- Git credential plugins (takes command input on stdin): https://git-scm.com/book/en/v2/Git-Tools-Credential-Storage
- (Other? I went through all of the CLI tools that I used to connect to infrastructure, but didn't find any good examples)

Reply:
Redirecting this conversation to https://github.com/kubernetes/kubernetes/pull/99310#discussion_r613462261.